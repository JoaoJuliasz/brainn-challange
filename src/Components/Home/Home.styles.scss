.home-container {
    height: 100%;
    display: flex;
    background: #e9e9e9;

    .loading-container {
        position: absolute;
        z-index: 45;
        background: #3d3d3d;
        width: 100%;
        height: 100%;
        display: flex;
        align-items: center;
        opacity: 0.6;
    }

    .select-type-container {
        width: 35%;
        height: 100%;

        @media (min-width: 860px) and (max-width: 950px) {
            width: 40%;
        }
    }

    .lottery-results-home-container {
        width: 65%;
        height: 100%;
    }

    @media (max-width: 860px) {
        flex-direction: column;

        .select-type-container {
            width: 100%;
        }

        .lottery-results-home-container {
            width: 100%;
        }
    }
}
