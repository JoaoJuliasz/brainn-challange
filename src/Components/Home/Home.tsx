import React, { useEffect, useState } from 'react'
// @ts-ignore
import instance from '../../config/config'
// @ts-ignore
import LotteryResults from '../LotteryResults/LotteryResults';
import SelectType from '../SelectType/SelectType';
import { Lotteries, Contest, LotteriesContests } from '../dataTypes/types'

import './Home.styles.scss';

type LotteriesT = Lotteries[]

const Home = () => {

    const [lotteries, setLotteries] = useState<LotteriesT>([] as LotteriesT)
    const [lotteriesContests, setLotteriesContests] = useState<LotteriesContests>([] as LotteriesContests)
    const [selectedLottery, setSelectedLottery] = useState<string>('0')
    const [contest, setContest] = useState<Contest>({} as Contest)
    const [loading, setLoading] = useState<boolean>(false)

    const getLotteriesValues = () => {
        instance.get('loterias')
            .then((res: any) => {
                const auxArr: LotteriesT = []
                res.data.map((lottery: { id: number, nome: string }): void => {
                    auxArr.push({ value: lottery.id, label: lottery?.nome })
                })
                setLotteries(auxArr)
            })
    }

    const getLotteriesContestsValues = () => {
        instance.get('loterias-concursos')
            .then((res: any) => setLotteriesContests(res.data))
    }

    useEffect(() => {
        getLotteriesValues()
        getLotteriesContestsValues()
    }, [])

    return (
        <div className="home-container">
            <div className="select-type-container">
                <SelectType lotteries={lotteries} selectedLottery={selectedLottery} setSelectedLottery={setSelectedLottery}
                    lotteriesContests={lotteriesContests} contest={contest} setContest={setContest} setLoading={setLoading} />
            </div>
            <div className="lottery-results-home-container">
                <LotteryResults contest={contest} loading={loading} />
            </div>
        </div>
    );
};

export default Home;