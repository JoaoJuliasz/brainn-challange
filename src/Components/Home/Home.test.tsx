import React from 'react'
// @ts-ignore
import { shallow } from 'enzyme'
import Home from './Home'

it('should return wrapper length', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper).toMatchSnapshot()
})