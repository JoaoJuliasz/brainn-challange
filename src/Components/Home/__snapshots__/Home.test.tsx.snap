// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return wrapper length 1`] = `
<div
  className="home-container"
>
  <div
    className="select-type-container"
  >
    <SelectType
      contest={Object {}}
      lotteries={Array []}
      lotteriesContests={Array []}
      selectedLottery="0"
      setContest={[Function]}
      setLoading={[Function]}
      setSelectedLottery={[Function]}
    />
  </div>
  <div
    className="lottery-results-home-container"
  >
    <LotteryResults
      contest={Object {}}
    />
  </div>
</div>
`;
